pkg/build/builder/docker.go

```diff
@@ -264,7 +264,10 @@ func (d *DockerBuilder) buildInfo() []dockerfile.KeyValue {
 // consume.
 func (d *DockerBuilder) buildLabels(dir string) []dockerfile.KeyValue {
 	labels := map[string]string{}
-	sourceInfo := d.git.GetInfo(dir)
+	sourceInfo := &s2iapi.SourceInfo{}
+	if d.build.Spec.Source.Git != nil {
```

Comment:
Shouldn't you test for the type? You can have a non-null Git struct that we just ignore.